### ISSUE TITLE
Questions/Issues about pekwm_panel

### ISSUE THREAD

**Issue by okraits** [pekdon/pekwm#95]
I have several questions/issues related to _pekwm_panel_. I wanted to do a PR to update the documentation but I want to discuss this first.

1. Could it be that pekwm_panel needs to be executed after pekwm? If I execute it before pekwm (from `~.xsession`),`%:WORKSPACE_NAME:` for example stays empty.
2. The panel documentation still mentions `WorkspaceNumber` and `ExternalData` although those widgets have been removed.
3. Is there currently any usecase for the `Required` unit of the `Size` parameter now that `WorkspaceNumber` has been removed?
4. To avoid unnecessary widgets, the `DateTime` widget could be replaced by a combination of the `Text` widget with a `Command`.
5. The documentation for `%:` is a little bit ambiguous as the first `:` appears both in the format string and in the variable.
6. Referencing environment variables in `Text` widgets: is it really `_%USER` and not `%_USER`? I couldn't get either of those to work.

**Comment by pekdon** [comment]
1. Does it update whenever you change a workspace after initially being empty? (It isn't set if starting from a clean X session but should get updated whenever it is updated by pekwm)

3. Not sure that it currently works, but for Text widgets where there are not variables (or environment variables only) Required should be the same as TextWidth of the value which is practical IMHO.

4. I intentionally kept the DateTime widget to avoid having to execute an external application every second if one wants the time to update on a per-second basis. In the name of efficiency it gets to stay around.

6. It is %_, just verified and this works for me:

```
  Text = "%_USER" {
    Size = "TextWidth _pekdon_"
  }
```

On 2 and 5, feel free to create a PR or I'll try to make a bit more clear in the next revision.

**Comment by pekdon** [comment]
Just pushed a fix removing some ExternalData and WorkspaceNumber references from the documentation and added Required calculation for the Text widget which works if no variables are referenced in the string. (Environment variables excluded)

**Comment by okraits** [comment]
>     1. Does it update whenever you change a workspace after initially being empty? (It isn't set if starting from a clean X session but should get updated whenever it is updated by pekwm)
No, it doesn't. It isn't empty when it is executed after pekwm, though.
>     3. I intentionally kept the DateTime widget to avoid having to execute an external application every second if one wants the time to update on a per-second basis. In the name of efficiency it gets to stay around. (or have a long running process)
Sounds reasonable :grinning: 
>     4. It is %_, just verified and this works for me:
> 
> 
> ```
>   Text = "%_USER" {
>     Size = "TextWidth _pekdon_"
>   }
> ```
Yes, that works for me too, I guess my TextWidth value was wrong.
> On 2 and 5, feel free to create a PR or I'll try to make a bit more clear in the next revision.

Sure, I will have a look.

**Comment by maoamid** [comment]
pekwm_panel has a size prop (width size) other than 100% ? 
I think it should be defined a prop that define a 'minimum' size relative to the sum of enabled widgets size. That also imply that his position needs to be defined for a smaller panel to be placed on screen (left / center / right)
